src/components/TeacherComponents/UnitList/helpers.tsx

```diff
@@ -0,0 +1,59 @@
+import { UnitListItemProps } from "../UnitListItem/UnitListItem";
+import { UnitOption } from "../UnitListOptionalityCard/UnitListOptionalityCard";
+
+import { CurrentPageItemsProps, UnitListProps } from "./UnitList";
+
+import isSlugLegacy from "@/utils/slugModifiers/isSlugLegacy";
+import {
+  SpecialistUnit,
+  SpecialistUnitListingData,
+} from "@/node-lib/curriculum-api-2023/queries/specialistUnitListing/specialistUnitListing.schema";
+import { UnitListingData } from "@/node-lib/curriculum-api-2023/queries/unitListing/unitListing.schema";
+
+export const isCurrentPageItems = (
+  u: CurrentPageItemsProps[] | SpecialistUnit[][],
+): u is CurrentPageItemsProps[] => {
+  return (u[0] as CurrentPageItemsProps)[0]?.keyStageSlug !== undefined;
+};
+
+export const isUnitListData = (
+  u: UnitListingData | SpecialistUnitListingData,
+): u is UnitListingData => {
+  return (u as UnitListingData).keyStageSlug !== undefined;
+};
+
+export const getPageItems = (
+  pageItems: CurrentPageItemsProps[] | SpecialistUnit[][],
+  legacy: boolean,
+) => {
+  return isCurrentPageItems(pageItems)
+    ? pageItems.filter((item) =>
+        legacy
+          ? isSlugLegacy(item[0]!.programmeSlug)
+          : !isSlugLegacy(item[0]!.programmeSlug),
+      )
+    : [];
+};
```

Comment:
🏄🏻 🤯 reading this broke me for a few reads. 

The argument name `legacy` obscures the intention (which I think is to let you filter out lessons with legacy slugs or vice versa).

The nested ternary hurt too. Could be refactored as an early return? 

```suggestion
export const getPageItems = (
  pageItems: CurrentPageItemsProps[] | SpecialistUnit[][],
  pickLegacyItems: boolean,
) => {
  if (!isCurrentPageItems(pageItems)) {
    return [];
  }

  return pageItems.filter((item) => {
    const isLegacy = isSlugLegacy(item[0]!.programmeSlug);
    return pickLegacyItems ? isLegacy : !isLegacy;
  });
};
```